### PR TITLE
Artifact2Artifact MRI demo + other small additions

### LINKED ISSUE #251: Simpler metrics should use more lightweight library
Simpler metrics such as `deepinv.loss.SSIM` should call a more lightweight library, e.g. `torchvision`, rather than `pyiqa` which has many dependencies such as `opencv` and `llvmlite`. Furthermore, `pyiqa`'s documentation isn't as well established and the API is quite clunky for simple metrics. 

Example: [pyiqa's SSIM](https://github.com/chaofengc/IQA-PyTorch/blob/main/pyiqa/archs/ssim_arch.py#L30) vs [torchmetric's SSIM](https://github.com/Lightning-AI/torchmetrics/blob/master/src/torchmetrics/image/ssim.py)

### PR DESCRIPTION
**Pull request by Andrewwango**
Losses
- Clean up Artifact2Artifact
- Demo for MRI reconstruction from sequential measurements using Artifact2Artifact.
- Closes #251 
- Add MSE and NMSE metrics
- Allow PSNR and SSIM to set max pixel on the fly (as they do in fastMRI)
- MOI loss accepts physics generator
- Add MOEI (multi-operator equivariant imaging)
- Add loss schedulers

Physics
- Add sequential MRI operator

Models
- Adds mixin support for temporal classes #292 + wrappers to adapt models to accept 2D+t input

### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [x] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: examples/self-supervised-learning/demo_artifact2artifact.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+# %%
+# Now define physics using this time-varying mask of shape [B,C,T,H,W]:
+#
+
+physics = dinv.physics.SequentialMRI(mask=mask)
+
+
+# %%
+# Let�s visualise the sequential measurements using a sample image (run
```
Comment (tachella) [inline examples/self-supervised-learning/demo_artifact2artifact.py:129]:
```suggestion
# Let's visualise the sequential measurements using a sample image (run
```

File: examples/self-supervised-learning/demo_artifact2artifact.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+print("Total acceleration:", (2 * 128 * 128) / mask.sum())
+
+
+# %%
+# Define model
+# ------------
+#
+# As a (static) reconstruction network, we use an unrolled network
+# (half-quadratic splitting) with a trainable denoising prior based on the
+# DnCNN architecture. See :meth:`deepinv.utils.demo.demo_mri_model` for
```
Comment (tachella) [inline examples/self-supervised-learning/demo_artifact2artifact.py:163]: same here regarding MoDL?

**Review by tachella — CHANGES_REQUESTED** [review]
Great PR, I really like the new MRI capabilities and the loss schedulers.

Besides a few esthetic comments, I'm wondering whether the MOEI loss is necessary or if we can simply add transforms to MOI? If you prefer to keep them separated, I would add a better description of the MOIEI in the docstring.

File: deepinv/loss/metric.py [hunk 3]
```diff
@@ -71,36 +76,95 @@ def forward(self, x, x_net, **kwargs):
...
+        return cal_mse(x_net, x)
+
+
+class NMSE(MSE):
+    r"""
+    Normalised Mean Squared Error metric.
+
+    Normalises MSE by the L2 norm of the ground truth ``x``.
+
+    TODO add various other normalisation methods from torchmetrics. See https://github.com/Lightning-AI/torchmetrics/pull/2442
```
[the remark this answers is not in the document]
Comment (Andrewwango) [reply deepinv/loss/metric.py:103]: Thanks @tachella, I was going to start a separate PR for more unified metrics (+ normalisation etc., for #299 ) so I can remove this from this for now and then do that after this is merged?

**Review by tachella — APPROVED** [review]

**Pull request merged**